storage/innobase/os/os0file.cc

```diff
@@ -9245,9 +9364,258 @@ Encryption::get_master_key(ulint* master_key_id,
 #endif
 }
 
+#ifndef UNIV_INNOCHECKSUM
+
+/** Fill the encryption information.
+@param[in]	key		encryption key
+@param[in]	iv		encryption iv
+@param[in,out]	encrypt_info	encryption information
+@param[in]	is_boot		if it's for bootstrap
+@return true if success */
+bool Encryption::fill_encryption_info(byte*	key,
+				      byte*	iv,
+				      byte*	encrypt_info,
+				      bool	is_boot)
+{
+	byte*			ptr;
+	lint			elen;
+	ulint			master_key_id;
+	byte*			master_key;
+	byte			key_info[ENCRYPTION_KEY_LEN * 2];
+	ulint			crc;
+	Version			version;
+
+	/* Get master key from key ring. For bootstrap, we use a default
+	master key which master_key_id is 0. */
+	if (is_boot) {
+		master_key_id = 0;
+		master_key = static_cast<byte*>(ut_zalloc_nokey(
```

Comment:
I'd rather use my_ allocator here too, so that it's always my_free regardless of is_boot